spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/jdbc/DataSourcePoolMetrics.java

```diff
@@ -108,10 +108,11 @@ public void bindTo(MeterRegistry registry) {
 
 		@Override
 		public DataSourcePoolMetadata getDataSourcePoolMetadata(DataSource dataSource) {
-			DataSourcePoolMetadata metadata = cache.get(dataSource);
+			int dataSourceKey = System.identityHashCode(dataSource);
```

Comment:
I don't think this is safe. Over the lifetime of the JVM, there's no guarantee that every `identityHashCode` will be unique.